### PR TITLE
Record serious/dangerous fault on tap & press & hold

### PR DESCRIPTION
**Pull request by szabolcs-keresztes**
Implementing the functionality to record serious or driving faults with tap or press or hold.
Also writing unit tests for this feature.

### CODE REVIEW

File: src/pages/test-report/components/competency/__tests__/competency.spec.ts [hunk 1]
```diff
@@ -85,38 +81,170 @@ describe('CompetencyComponent', () => {
...
         component.competency = Competencies.controlsSteering;
         expect(component.getLabel()).toBe('Steering');
       });
     });
+
     describe('recordFault', () => {
       it('should dispatch a ADD_DRIVING_FAULT action', () => {
         component.competency = Competencies.controlsSteering;
 
+        const storeDispatchSpy = spyOn(store$, 'dispatch');
```
Comment (Jason-Pullen) [inline src/pages/test-report/components/competency/__tests__/competency.spec.ts]: no need to assign the result of spyOn to storeDispatchSpy, you can just call expect(store$.dispatch).toHaveBeenCalled
multiple places

File: src/pages/test-report/components/competency/competency.ts [hunk 2]
```diff
@@ -126,6 +126,43 @@ export class CompetencyComponent {
 
   getLabel = (): string => competencyLabels[this.competency];
 
+  recordSerious = (): void => {
+    this.store$.dispatch(new AddSeriousFault(this.competency));
```
Comment (Jason-Pullen) [inline src/pages/test-report/components/competency/competency.ts]: are recordSerious and recordDangerous used?

File: src/pages/test-report/components/competency/__tests__/competency.spec.ts (hunk 1 above)
Comment (szabolcs-keresztes) [reply src/pages/test-report/components/competency/__tests__/competency.spec.ts]: You are right, however being a bit more explicit always improves readability.

File: src/pages/test-report/components/competency/competency.ts (hunk 2 above)
Comment (szabolcs-keresztes) [reply src/pages/test-report/components/competency/competency.ts]: Good point, it was just experimental code. Removed them now.

**Pull request merged**